{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "root_dir = os.path.abspath(os.path.join(os.getcwd(), \"../\"))\n",
    "os.chdir(root_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import io\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "import pandas as pd\n",
    "sns.set_style(\"whitegrid\")\n",
    "sns.set_context(\"paper\", font_scale=1.4)\n",
    "# sns.set_palette(\"Set2\")\n",
    "import functions\n",
    "\n",
    "from multiprocessing import Pool\n",
    "from functools import partial\n",
    "\n",
    "from sklearn.metrics import roc_curve, auc\n",
    "from sklearn.metrics import precision_recall_curve\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.metrics import matthews_corrcoef\n",
    "\n",
    "# from ipywidgets import interact\n",
    "# import ipywidgets as widgets\n",
    "from scipy.signal import savgol_filter\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "\n",
    "\n",
    "# Kyte & Doolittle index of hydrophobicity\n",
    "# J. Mol. Biol. 157:105-132(1982).\n",
    "kd = {\"A\": 1.8, \"R\": -4.5, \"N\": -3.5, \"D\": -3.5, \"C\": 2.5,\n",
    "      \"Q\": -3.5, \"E\": -3.5, \"G\": -0.4, \"H\": -3.2, \"I\": 4.5,\n",
    "      \"L\": 3.8, \"K\": -3.9, \"M\": 1.9, \"F\": 2.8, \"P\": -1.6,\n",
    "      \"S\": -0.8, \"T\": -0.7, \"W\": -0.9, \"Y\": -1.3, \"V\": 4.2}\n",
    "\n",
    "#Sort kd\n",
    "kd = {k: v for k, v in sorted(kd.items(), key=lambda item: item[1])}\n",
    "\n",
    "# Flexibility\n",
    "# Normalized flexibility parameters (B-values), average\n",
    "# Vihinen M., Torkkila E., Riikonen P. Proteins. 19(2):141-9(1994).\n",
    "flexibilities_vih = {\"A\": 0.984, \"C\": 0.906, \"E\": 1.094, \"D\": 1.068,\n",
    "\"G\": 1.031, \"F\": 0.915, \"I\": 0.927, \"H\": 0.950,\n",
    "\"K\": 1.102, \"M\": 0.952, \"L\": 0.935, \"N\": 1.048,\n",
    "\"Q\": 1.037, \"P\": 1.049, \"S\": 1.046, \"R\": 1.008,\n",
    "\"T\": 0.997, \"W\": 0.904, \"V\": 0.931, \"Y\": 0.929}\n",
    "\n",
    "def flex(seq):\n",
    "    seq = seq.replace('U', 'C')\n",
    "    if len(seq) == 0:\n",
    "        return 0\n",
    "    else:\n",
    "        return [flexibilities_vih[i] for i in seq]\n",
    "\n",
    "def gravy(seq):\n",
    "    seq = seq.replace('U', 'C')\n",
    "    if len(seq) == 0:\n",
    "        return 0\n",
    "    else:\n",
    "        return np.mean([kd[i] for i in seq])\n",
    "\n",
    "def hydrop(seq):\n",
    "    seq = seq.replace('U', 'C')\n",
    "    return [kd[i] for i in seq]\n",
    "\n",
    "def filt(seq, w=15, p=2):\n",
    "    seq = seq.replace('U', 'C')\n",
    "    aa = hydrop(seq)\n",
    "    return savgol_filter(aa, w, p)\n",
    "\n",
    "aa = [k for k, v in kd.items()]\n",
    "\n",
    "\n",
    "def progress(iteration, total, message=None):\n",
    "    '''Simple progressbar\n",
    "    '''\n",
    "    if message is None:\n",
    "        message = ''\n",
    "    bars_string = int(float(iteration) / float(total) * 50.)\n",
    "    print(\"\\r|%-50s| %d%% (%s/%s) %s \"% ('█'*bars_string+ \"░\" * \\\n",
    "                                     (50 - bars_string), float(iteration)/\\\n",
    "                                     float(total) * 100, iteration, total, \\\n",
    "                                     message), end='\\r', flush=True)\n",
    "\n",
    "    if iteration == total:\n",
    "        print('\\nCompleted!')\n",
    "\n",
    "\n",
    "\n",
    "def make_feat(seq):\n",
    "    aa_list = 'RKNDCEVIYFWL' + 'QP'\n",
    "    h = hydrop(seq)\n",
    "    swi = functions.solubility_score(seq)\n",
    "    fl = flex(seq)\n",
    "    aa_counts = [seq.count(i) for i in aa_list]\n",
    "    return list(savgol_filter(h, 15, 2)) +  list(savgol_filter(swi, 15, 2)) + fl + aa_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>All</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>Signal peptide</th>\n",
       "      <th>...</th>\n",
       "      <th>Cross-reference (Pfam)</th>\n",
       "      <th>Biotechnological use</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Domain [FT]</th>\n",
       "      <th>Cross-reference (InterPro)</th>\n",
       "      <th>Cross-reference (PROSITE)</th>\n",
       "      <th>Cleavage</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Label</th>\n",
       "      <th>Aligned</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5947</th>\n",
       "      <td>O35129</td>\n",
       "      <td>sp|O35129|PHB2_MOUSE Prohibitin-2 OS=Mus muscu...</td>\n",
       "      <td>MAQNLKDLAGRLPAGPRGMGTALKLLLGAGAVAYGVRESVFTVEGG...</td>\n",
       "      <td>PHB2_MOUSE</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Prohibitin-2 (B-cell receptor-associated prote...</td>\n",
       "      <td>Phb2 Bap Bcap37 Rea</td>\n",
       "      <td>Mus musculus (Mouse)</td>\n",
       "      <td>299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PF01145;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IPR001107;IPR036013;IPR000163;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O35129|EUKARYA|NO_SP|1</td>\n",
       "      <td>0</td>\n",
       "      <td>AQNLKDLAGRLPAGPRGMGTALKLLLGAGAVAYGVRESVFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6252</th>\n",
       "      <td>F4IE66</td>\n",
       "      <td>sp|F4IE66|PRP22_ARATH Pre-mRNA-splicing factor...</td>\n",
       "      <td>MPSMAQGELKSFVQNSRPNPKSPTVSPFSMRQKIAEHRRSLPIASV...</td>\n",
       "      <td>PRP22_ARATH</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Pre-mRNA-splicing factor ATP-dependent RNA hel...</td>\n",
       "      <td>RID1 At1g26370 T1K7.25</td>\n",
       "      <td>Arabidopsis thaliana (Mouse-ear cress)</td>\n",
       "      <td>717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PF04408;PF00271;PF07717;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MOTIF 162..165;  /note=\"DEAH box\";  /evidence=...</td>\n",
       "      <td>DOMAIN 51..256;  /note=\"Helicase ATP-binding\";...</td>\n",
       "      <td>IPR003593;IPR011709;IPR007502;IPR014001;IPR001...</td>\n",
       "      <td>PS51192;PS51194;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F4IE66|EUKARYA|NO_SP|0</td>\n",
       "      <td>0</td>\n",
       "      <td>PSMAQGELKSFVQNSRPNPKSPTVSPFSMRQKIAEHRRSLP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Entry                                                All  \\\n",
       "5947  O35129  sp|O35129|PHB2_MOUSE Prohibitin-2 OS=Mus muscu...   \n",
       "6252  F4IE66  sp|F4IE66|PRP22_ARATH Pre-mRNA-splicing factor...   \n",
       "\n",
       "                                                Protein   Entry name  \\\n",
       "5947  MAQNLKDLAGRLPAGPRGMGTALKLLLGAGAVAYGVRESVFTVEGG...   PHB2_MOUSE   \n",
       "6252  MPSMAQGELKSFVQNSRPNPKSPTVSPFSMRQKIAEHRRSLPIASV...  PRP22_ARATH   \n",
       "\n",
       "        Status                                      Protein names  \\\n",
       "5947  reviewed  Prohibitin-2 (B-cell receptor-associated prote...   \n",
       "6252  reviewed  Pre-mRNA-splicing factor ATP-dependent RNA hel...   \n",
       "\n",
       "                  Gene names                                Organism  Length  \\\n",
       "5947     Phb2 Bap Bcap37 Rea                    Mus musculus (Mouse)     299   \n",
       "6252  RID1 At1g26370 T1K7.25  Arabidopsis thaliana (Mouse-ear cress)     717   \n",
       "\n",
       "     Signal peptide  ...    Cross-reference (Pfam) Biotechnological use  \\\n",
       "5947            NaN  ...                  PF01145;                  NaN   \n",
       "6252            NaN  ...  PF04408;PF00271;PF07717;                  NaN   \n",
       "\n",
       "                                                  Motif  \\\n",
       "5947                                                NaN   \n",
       "6252  MOTIF 162..165;  /note=\"DEAH box\";  /evidence=...   \n",
       "\n",
       "                                            Domain [FT]  \\\n",
       "5947                                                NaN   \n",
       "6252  DOMAIN 51..256;  /note=\"Helicase ATP-binding\";...   \n",
       "\n",
       "                             Cross-reference (InterPro)  \\\n",
       "5947                     IPR001107;IPR036013;IPR000163;   \n",
       "6252  IPR003593;IPR011709;IPR007502;IPR014001;IPR001...   \n",
       "\n",
       "     Cross-reference (PROSITE) Cleavage               Accession Label  \\\n",
       "5947                       NaN      NaN  O35129|EUKARYA|NO_SP|1     0   \n",
       "6252          PS51192;PS51194;      NaN  F4IE66|EUKARYA|NO_SP|0     0   \n",
       "\n",
       "                                        Aligned  \n",
       "5947  AQNLKDLAGRLPAGPRGMGTALKLLLGAGAVAYGVRESVFT  \n",
       "6252  PSMAQGELKSFVQNSRPNPKSPTVSPFSMRQKIAEHRRSLP  \n",
       "\n",
       "[2 rows x 42 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ = pd.read_pickle('results/signalp5_train_all.pkl.gz')\n",
    "#Shuffle data\n",
    "df = df_.sample(frac=1, random_state=12345)\n",
    "df.head(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|██████████████████████████████████████████████████| 100% (5/5)  \n",
      "Completed!\n"
     ]
    }
   ],
   "source": [
    "\n",
    "ll = np.arange(5)\n",
    "\n",
    "df['Feat'] = df['Protein'].str[:30].apply(make_feat)\n",
    "\n",
    "df_s = []\n",
    "\n",
    "for q in range(5):\n",
    "    df_s.append(df[df['Accession'].apply(lambda x: '|{}'.format(q) in x)].copy())\n",
    "\n",
    "\n",
    "cv_results = pd.DataFrame(columns=['MCC', 'Classifier', 'Threshold'], index=np.arange(1, 6))\n",
    "ll = np.arange(5)\n",
    "\n",
    "for j in range(5):\n",
    "    test_idx = j\n",
    "    train_idx = list(set(ll) - set([j]))\n",
    "\n",
    "    test = df_s[j]\n",
    "    train = pd.concat([df_s[i] for i in train_idx])\n",
    "\n",
    "    train = train.reset_index(drop=True)\n",
    "    train = train.sample(frac=1, random_state=12345).copy()\n",
    "\n",
    "\n",
    "    X_train = [i for i in train['Feat']]\n",
    "    y_train = train['Label'].values\n",
    "\n",
    "\n",
    "    X_test = [i for i in test['Feat']]\n",
    "    y_test = test['Label'].values\n",
    "\n",
    "\n",
    "    clf = RandomForestClassifier(random_state=12345, ) #n_jobs=-1\n",
    "    clf.fit(X_train, y_train)\n",
    "    cv_results.loc[j+1]['Classifier'] = clf\n",
    "\n",
    "    preds = [i[1] for i in clf.predict_proba(X_test)]\n",
    "    df_s[j]['SP_Prediction'] = preds\n",
    "\n",
    "    mcc = []\n",
    "    threshold = []\n",
    "\n",
    "    for i, v in enumerate(np.arange(0.01, 0.99, 0.01)):\n",
    "        y_pred = df_s[j]['SP_Prediction'].apply(lambda x: 0 if x < v else 1)\n",
    "        y_true = df_s[j]['Label']\n",
    "        mcc.append(matthews_corrcoef(y_true, y_pred))\n",
    "\n",
    "\n",
    "    cv_results.loc[j+1]['MCC'] = max(mcc)\n",
    "    cv_results.loc[j+1]['Threshold'] = np.arange(0.01, 0.99, 0.01)[np.argmax(mcc)]\n",
    "\n",
    "    progress(j+1, 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.906245550741527"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "np.median(cv_results['MCC'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9121749803366043"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# np.median(cv_results['MCC'])\n",
    "# 0.9121749803366043"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Export scored dataframe\n",
    "final_df = pd.concat(df_s)\n",
    "final_df.to_pickle('results/SP_NOSP_scored_QP.pkl.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "cv_results.to_pickle('results/S.pkl.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MCC</th>\n",
       "      <th>Classifier</th>\n",
       "      <th>Threshold</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.922195</td>\n",
       "      <td>(DecisionTreeClassifier(max_features='auto', r...</td>\n",
       "      <td>0.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.919256</td>\n",
       "      <td>(DecisionTreeClassifier(max_features='auto', r...</td>\n",
       "      <td>0.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.900328</td>\n",
       "      <td>(DecisionTreeClassifier(max_features='auto', r...</td>\n",
       "      <td>0.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.888793</td>\n",
       "      <td>(DecisionTreeClassifier(max_features='auto', r...</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.906246</td>\n",
       "      <td>(DecisionTreeClassifier(max_features='auto', r...</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        MCC                                         Classifier Threshold\n",
       "1  0.922195  (DecisionTreeClassifier(max_features='auto', r...      0.48\n",
       "2  0.919256  (DecisionTreeClassifier(max_features='auto', r...      0.53\n",
       "3  0.900328  (DecisionTreeClassifier(max_features='auto', r...      0.42\n",
       "4  0.888793  (DecisionTreeClassifier(max_features='auto', r...       0.5\n",
       "5  0.906246  (DecisionTreeClassifier(max_features='auto', r...       0.5"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cv_results = pd.read_pickle('results/S.pkl.gz')\n",
    "cv_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
