{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "import seaborn as sns\n",
    "sns.set_context(\"paper\", font_scale=1.6)\n",
    "\n",
    "def fasta_reader(file):\n",
    "    '''Converts .fasta to a pandas dataframe with accession as index\n",
    "    and sequence in a column 'sequence'\n",
    "    '''\n",
    "    fasta_df = pd.read_csv(file, sep='>', lineterminator='>', header=None)\n",
    "    fasta_df[['Accession', 'Sequence']] = fasta_df[0].str.split('\\n', 1, \\\n",
    "                                        expand=True)\n",
    "    fasta_df['Accession'] = fasta_df['Accession']\n",
    "    fasta_df['Sequence'] = fasta_df['Sequence'].replace('\\n', '', regex=True).\\\n",
    "                            astype(str).str.upper().replace('U', 'T')\n",
    "    total_seq = fasta_df.shape[0]\n",
    "    fasta_df.drop(0, axis=1, inplace=True)\n",
    "    fasta_df = fasta_df[fasta_df.Sequence != '']\n",
    "    fasta_df = fasta_df[fasta_df.Sequence != 'NONE']\n",
    "    final_df = fasta_df.dropna()\n",
    "    remained_seq = final_df.shape[0]\n",
    "    if total_seq != remained_seq:\n",
    "        print(\"{} sequences were removed due to inconsistencies in\"\n",
    "                      \"provided file.\".format(total_seq-remained_seq))\n",
    "    return final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "rev_df_ = fasta_reader('../../Signal_Manuscript/Uniprot_all/Fungi/uniprot-reviewed yes+taxonomy 4751.fasta.gz')\n",
    "\n",
    "rev_df_['Entry'] = rev_df_['Accession'].apply(lambda x: str(x).split('|')[1])\n",
    "rev_df_ = rev_df_.rename(columns={'Sequence':'Protein'})\n",
    "rev_df = rev_df_[~rev_df_.Protein.str[:75].str.contains('B|J|O|X|Z')].copy()\n",
    "\n",
    "rev_ = rev_df[['Protein', 'Entry']].copy()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(34901, 37)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot = pd.read_csv('../../Signal_Manuscript/Uniprot_all/Fungi/uniprot-reviewed yes+taxonomy 4751.tab.gz', sep='\\t')\n",
    "reviewed = uniprot.merge(rev_, on='Entry')\n",
    "reviewed.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def predict(obj):\n",
    "    res = obj.predict()\n",
    "    try:\n",
    "        f = obj.fungi()\n",
    "    except Exception:\n",
    "        f = ['Not a Signal peptide']\n",
    "        \n",
    "    try:\n",
    "        t = obj.toxin()\n",
    "    except Exception:\n",
    "        t = ['Not a Signal peptide']\n",
    "\n",
    "    return res +  f + t\n",
    "\n",
    "def progress(iteration, total, message=None):\n",
    "    '''Simple progressbar\n",
    "    '''\n",
    "    if message is None:\n",
    "        message = ''\n",
    "    bars_string = int(float(iteration) / float(total) * 50.)\n",
    "    print(\"\\r|%-50s| %d%% (%s/%s) %s \"% ('█'*bars_string+ \"░\" * \\\n",
    "                                     (50 - bars_string), float(iteration)/\\\n",
    "                                     float(total) * 100, iteration, total, \\\n",
    "                                     message), end='\\r', flush=True)\n",
    "\n",
    "    if iteration == total:\n",
    "        print('\\nCompleted!')\n",
    "\n",
    "\n",
    "from multiprocessing import Pool, cpu_count\n",
    "import time\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "from razor import RAZOR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqs_obj = [RAZOR(x, 60) for x in reviewed.Protein]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|██████████████████████████████████████████████████| 100% (34901/34901)  \n",
      "Completed!\n",
      "--- 125.8046190738678 seconds --- for 34901 sequences.\n",
      " 277.422021 sequences per second\n",
      "\n",
      " Completed!\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "start_time = time.time()\n",
    "pools = Pool(cpu_count()//4)\n",
    "results = []\n",
    "for res in pools.imap(predict, seqs_obj):\n",
    "    results.append(res)\n",
    "    progress(len(results), len(seqs_obj))\n",
    "pools.close()\n",
    "pools.join()\n",
    "print(\"--- %s seconds --- for %d sequences.\\n %f sequences per second\"%(time.time() - start_time, reviewed.shape[0], \\\n",
    "                                             reviewed.shape[0]/(time.time() - start_time)))\n",
    "print('\\n Completed!')\n",
    "\n",
    "\n",
    "\n",
    "reviewed['Preds'] = results\n",
    "reviewed.to_pickle('uniprot_reviewed_scored_fungi.pkl.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reviewed seqs\n",
    "\n",
    "df = pd.read_pickle('uniprot_reviewed_scored_fungi.pkl.gz')\n",
    "cols = ['SP', 'Y', 'C', 'All_Cleavage', 'Fungi_Preds', 'Fungi_prob', 'Toxin_Preds', 'Toxin_prob']\n",
    "\n",
    "tox_fung = df[df['Preds'].apply(len) == 8].copy()\n",
    "\n",
    "tox_fung[cols] = pd.DataFrame(tox_fung.Preds.tolist(), index= tox_fung.index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reviewed set has 473 probable fungi.\n"
     ]
    }
   ],
   "source": [
    "#Take those with 5 models returning True for toxins\n",
    "probable_sp = tox_fung[tox_fung['SP'].apply(lambda x: (x == True).sum()) == 5].copy()\n",
    "\n",
    "probable_fungi = probable_sp[probable_sp['Fungi_Preds'].apply(lambda x: (x ==True).sum()) == 5].copy()\n",
    "print('Reviewed set has {} probable fungi.'.format(probable_fungi.shape[0]))\n",
    "# print('====\\n')\n",
    "# print(','.join([i for i in probable_fungi.Entry.values]))\n",
    "# print('====\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# See if any from training is in the predicted sequences\n",
    "# and remove them\n",
    "\n",
    "train = pd.read_pickle('../results/fungi_nonfungi.pkl.gz')\n",
    "train = train[train.Label == 1]\n",
    "\n",
    "all_train_atp_seq = set(list(train.Entry.values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(121, 42)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>Signal peptide</th>\n",
       "      <th>Taxonomic lineage IDs</th>\n",
       "      <th>Date of creation</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Preds</th>\n",
       "      <th>SP</th>\n",
       "      <th>Y</th>\n",
       "      <th>C</th>\n",
       "      <th>All_Cleavage</th>\n",
       "      <th>Fungi_Preds</th>\n",
       "      <th>Fungi_prob</th>\n",
       "      <th>Toxin_Preds</th>\n",
       "      <th>Toxin_prob</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>D4ARV2</td>\n",
       "      <td>AIM6_ARTBC</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Altered inheritance of mitochondria protein 6 ...</td>\n",
       "      <td>ARB_06966</td>\n",
       "      <td>Arthroderma benhamiae (strain ATCC MYA-4681 / ...</td>\n",
       "      <td>314</td>\n",
       "      <td>SIGNAL 1..21;  /evidence=\"ECO:0000255\"</td>\n",
       "      <td>663331</td>\n",
       "      <td>2015-10-14</td>\n",
       "      <td>...</td>\n",
       "      <td>MKSSILASAAILAASLEPVAALSEPLQNILINTDRSPIYRYPTDVT...</td>\n",
       "      <td>[[True, True, True, True, True], 0.73498299300...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.734983</td>\n",
       "      <td>0.74</td>\n",
       "      <td>[21, 21]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.53, 0.37, 0.43, 0.32, 0.46]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.15, 0.11, 0.12, 0.1, 0.16]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Q0CF74</td>\n",
       "      <td>AZPB3_ASPTN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>FAD-linked oxidoreductase ATEG_07660 (EC 1.-.-...</td>\n",
       "      <td>ATEG_07660</td>\n",
       "      <td>Aspergillus terreus (strain NIH 2624 / FGSC A1...</td>\n",
       "      <td>481</td>\n",
       "      <td>SIGNAL 1..17;  /evidence=\"ECO:0000255\"</td>\n",
       "      <td>341663</td>\n",
       "      <td>2020-06-17</td>\n",
       "      <td>...</td>\n",
       "      <td>MLRQLYQSLALVAAVHATGIFANGVDIESLFGPYLSPGTEIAEPTD...</td>\n",
       "      <td>[[True, True, True, True, True], 0.65482822174...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.654828</td>\n",
       "      <td>0.71</td>\n",
       "      <td>[22]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.24, 0.26, 0.37, 0.24, 0.26]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.04, 0.08, 0.06, 0.05, 0.07]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Entry   Entry name    Status  \\\n",
       "9   D4ARV2   AIM6_ARTBC  reviewed   \n",
       "66  Q0CF74  AZPB3_ASPTN  reviewed   \n",
       "\n",
       "                                        Protein names  Gene names  \\\n",
       "9   Altered inheritance of mitochondria protein 6 ...   ARB_06966   \n",
       "66  FAD-linked oxidoreductase ATEG_07660 (EC 1.-.-...  ATEG_07660   \n",
       "\n",
       "                                             Organism  Length  \\\n",
       "9   Arthroderma benhamiae (strain ATCC MYA-4681 / ...     314   \n",
       "66  Aspergillus terreus (strain NIH 2624 / FGSC A1...     481   \n",
       "\n",
       "                            Signal peptide  Taxonomic lineage IDs  \\\n",
       "9   SIGNAL 1..21;  /evidence=\"ECO:0000255\"                 663331   \n",
       "66  SIGNAL 1..17;  /evidence=\"ECO:0000255\"                 341663   \n",
       "\n",
       "   Date of creation  ...                                            Protein  \\\n",
       "9        2015-10-14  ...  MKSSILASAAILAASLEPVAALSEPLQNILINTDRSPIYRYPTDVT...   \n",
       "66       2020-06-17  ...  MLRQLYQSLALVAAVHATGIFANGVDIESLFGPYLSPGTEIAEPTD...   \n",
       "\n",
       "                                                Preds  \\\n",
       "9   [[True, True, True, True, True], 0.73498299300...   \n",
       "66  [[True, True, True, True, True], 0.65482822174...   \n",
       "\n",
       "                                SP         Y     C All_Cleavage  \\\n",
       "9   [True, True, True, True, True]  0.734983  0.74     [21, 21]   \n",
       "66  [True, True, True, True, True]  0.654828  0.71         [22]   \n",
       "\n",
       "                       Fungi_Preds                      Fungi_prob  \\\n",
       "9   [True, True, True, True, True]  [0.53, 0.37, 0.43, 0.32, 0.46]   \n",
       "66  [True, True, True, True, True]  [0.24, 0.26, 0.37, 0.24, 0.26]   \n",
       "\n",
       "                            Toxin_Preds                      Toxin_prob  \n",
       "9   [False, False, False, False, False]   [0.15, 0.11, 0.12, 0.1, 0.16]  \n",
       "66  [False, False, False, False, False]  [0.04, 0.08, 0.06, 0.05, 0.07]  \n",
       "\n",
       "[2 rows x 46 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filtered__ = df_uniprot[df_uniprot['Entry'].apply(lambda x: x not in all_train_atp_seq)].copy()\n",
    "filtered__ = probable_fungi[probable_fungi['Entry'].apply(lambda x: x not in all_train_atp_seq)].copy()\n",
    "filtered__.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_tm(annot):\n",
    "    '''\n",
    "    Find transmembrane annotation if it is outside \n",
    "    first 50 residues\n",
    "    '''\n",
    "    annot = str(annot)\n",
    "    if annot != 'NaN':\n",
    "        try:\n",
    "            if int(annot.split(';')[0].split(' ')[1].split('..')[0]) < 50:\n",
    "                return True\n",
    "            else:\n",
    "                return False\n",
    "        except Exception:\n",
    "#             print(annot)\n",
    "            return False\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>Signal peptide</th>\n",
       "      <th>Taxonomic lineage IDs</th>\n",
       "      <th>Date of creation</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Preds</th>\n",
       "      <th>SP</th>\n",
       "      <th>Y</th>\n",
       "      <th>C</th>\n",
       "      <th>All_Cleavage</th>\n",
       "      <th>Fungi_Preds</th>\n",
       "      <th>Fungi_prob</th>\n",
       "      <th>Toxin_Preds</th>\n",
       "      <th>Toxin_prob</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Q01302</td>\n",
       "      <td>CCG6_NEUCR</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Clock-controlled protein 6</td>\n",
       "      <td>ccg-6 B8P8.100 NCU01418</td>\n",
       "      <td>Neurospora crassa (strain ATCC 24698 / 74-OR23...</td>\n",
       "      <td>142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>367110</td>\n",
       "      <td>1997-11-01</td>\n",
       "      <td>...</td>\n",
       "      <td>MKFSAAAVLAAAAGAHAWSNVTYTTEIVTAVTTYCPGPTEITHGGN...</td>\n",
       "      <td>[[True, True, True, True, True], 0.71847059786...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.718471</td>\n",
       "      <td>0.86</td>\n",
       "      <td>[17]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.34, 0.54, 0.43, 0.35, 0.31]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.1, 0.11, 0.09, 0.08, 0.09]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>P87167</td>\n",
       "      <td>BUT2_SCHPO</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Uba3-binding protein but2</td>\n",
       "      <td>but2 SPBC3D6.02</td>\n",
       "      <td>Schizosaccharomyces pombe (strain 972 / ATCC 2...</td>\n",
       "      <td>390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>284812</td>\n",
       "      <td>1998-07-15</td>\n",
       "      <td>...</td>\n",
       "      <td>MQLLNSFLGFAASIAVLASSADAAPTLYKRKSKSNTNTAKSVAILD...</td>\n",
       "      <td>[[True, True, True, True, True], 0.74458041875...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.744580</td>\n",
       "      <td>0.68</td>\n",
       "      <td>[23]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.48, 0.27, 0.45, 0.5, 0.54]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.06, 0.11, 0.1, 0.06, 0.12]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7939</th>\n",
       "      <td>G2TRT4</td>\n",
       "      <td>NEW25_SCHPO</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>SAP domain-containing new25</td>\n",
       "      <td>new25 SPCC330.21</td>\n",
       "      <td>Schizosaccharomyces pombe (strain 972 / ATCC 2...</td>\n",
       "      <td>89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>284812</td>\n",
       "      <td>2012-04-18</td>\n",
       "      <td>...</td>\n",
       "      <td>MKIANVLTLSFAAIIAASSLAVVVSPITDNSRETASSSISQSPPSQ...</td>\n",
       "      <td>[[True, True, True, True, True], 0.65268675488...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.652687</td>\n",
       "      <td>0.71</td>\n",
       "      <td>[21]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.68, 0.58, 0.49, 0.68, 0.49]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.15, 0.21, 0.13, 0.22, 0.14]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31525</th>\n",
       "      <td>A0A0B4GDU5</td>\n",
       "      <td>ARP1_METBS</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Scytalone dehydratase-like protein Arp1 (EC 4....</td>\n",
       "      <td>Arp1 MBR_03976</td>\n",
       "      <td>Metarhizium brunneum (strain ARSEF 3297)</td>\n",
       "      <td>744</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276141</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>...</td>\n",
       "      <td>MGWNQTFLFLAFAATAASSLVGSGTSLQLNGIDYFVSPFSQGKVTN...</td>\n",
       "      <td>[[True, True, True, True, True], 0.83426614458...</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>0.834266</td>\n",
       "      <td>0.84</td>\n",
       "      <td>[19, 19]</td>\n",
       "      <td>[True, True, True, True, True]</td>\n",
       "      <td>[0.39, 0.4, 0.34, 0.28, 0.26]</td>\n",
       "      <td>[False, False, False, False, False]</td>\n",
       "      <td>[0.03, 0.04, 0.05, 0.07, 0.01]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Entry   Entry name    Status  \\\n",
       "264        Q01302   CCG6_NEUCR  reviewed   \n",
       "614        P87167   BUT2_SCHPO  reviewed   \n",
       "7939       G2TRT4  NEW25_SCHPO  reviewed   \n",
       "31525  A0A0B4GDU5   ARP1_METBS  reviewed   \n",
       "\n",
       "                                           Protein names  \\\n",
       "264                           Clock-controlled protein 6   \n",
       "614                            Uba3-binding protein but2   \n",
       "7939                         SAP domain-containing new25   \n",
       "31525  Scytalone dehydratase-like protein Arp1 (EC 4....   \n",
       "\n",
       "                    Gene names  \\\n",
       "264    ccg-6 B8P8.100 NCU01418   \n",
       "614            but2 SPBC3D6.02   \n",
       "7939          new25 SPCC330.21   \n",
       "31525           Arp1 MBR_03976   \n",
       "\n",
       "                                                Organism  Length  \\\n",
       "264    Neurospora crassa (strain ATCC 24698 / 74-OR23...     142   \n",
       "614    Schizosaccharomyces pombe (strain 972 / ATCC 2...     390   \n",
       "7939   Schizosaccharomyces pombe (strain 972 / ATCC 2...      89   \n",
       "31525           Metarhizium brunneum (strain ARSEF 3297)     744   \n",
       "\n",
       "      Signal peptide  Taxonomic lineage IDs Date of creation  ...  \\\n",
       "264              NaN                 367110       1997-11-01  ...   \n",
       "614              NaN                 284812       1998-07-15  ...   \n",
       "7939             NaN                 284812       2012-04-18  ...   \n",
       "31525            NaN                1276141       2018-12-05  ...   \n",
       "\n",
       "                                                 Protein  \\\n",
       "264    MKFSAAAVLAAAAGAHAWSNVTYTTEIVTAVTTYCPGPTEITHGGN...   \n",
       "614    MQLLNSFLGFAASIAVLASSADAAPTLYKRKSKSNTNTAKSVAILD...   \n",
       "7939   MKIANVLTLSFAAIIAASSLAVVVSPITDNSRETASSSISQSPPSQ...   \n",
       "31525  MGWNQTFLFLAFAATAASSLVGSGTSLQLNGIDYFVSPFSQGKVTN...   \n",
       "\n",
       "                                                   Preds  \\\n",
       "264    [[True, True, True, True, True], 0.71847059786...   \n",
       "614    [[True, True, True, True, True], 0.74458041875...   \n",
       "7939   [[True, True, True, True, True], 0.65268675488...   \n",
       "31525  [[True, True, True, True, True], 0.83426614458...   \n",
       "\n",
       "                                   SP         Y     C All_Cleavage  \\\n",
       "264    [True, True, True, True, True]  0.718471  0.86         [17]   \n",
       "614    [True, True, True, True, True]  0.744580  0.68         [23]   \n",
       "7939   [True, True, True, True, True]  0.652687  0.71         [21]   \n",
       "31525  [True, True, True, True, True]  0.834266  0.84     [19, 19]   \n",
       "\n",
       "                          Fungi_Preds                      Fungi_prob  \\\n",
       "264    [True, True, True, True, True]  [0.34, 0.54, 0.43, 0.35, 0.31]   \n",
       "614    [True, True, True, True, True]   [0.48, 0.27, 0.45, 0.5, 0.54]   \n",
       "7939   [True, True, True, True, True]  [0.68, 0.58, 0.49, 0.68, 0.49]   \n",
       "31525  [True, True, True, True, True]   [0.39, 0.4, 0.34, 0.28, 0.26]   \n",
       "\n",
       "                               Toxin_Preds                      Toxin_prob  \n",
       "264    [False, False, False, False, False]   [0.1, 0.11, 0.09, 0.08, 0.09]  \n",
       "614    [False, False, False, False, False]   [0.06, 0.11, 0.1, 0.06, 0.12]  \n",
       "7939   [False, False, False, False, False]  [0.15, 0.21, 0.13, 0.22, 0.14]  \n",
       "31525  [False, False, False, False, False]  [0.03, 0.04, 0.05, 0.07, 0.01]  \n",
       "\n",
       "[4 rows x 46 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "sp_fungi = filtered__[~filtered__['Transmembrane'].apply(find_tm)].copy()\n",
    "\n",
    "sp_fungi[sp_fungi['Signal peptide'].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Q01302\n",
      "MKFSAAAVLAAAAGAHAWSNVTYTTEIVTAVTTYCPGPTEITHGGNTYTVTEATTLTISDCPCTVTKPIITTSSVICHSCTGYVNSTIPAPTSAGSVGTGSAPAVVTPTVSPSEVPTAGAGKAAALSGAGLVGVLGLAAILL\n",
      ">P87167\n",
      "MQLLNSFLGFAASIAVLASSADAAPTLYKRKSKSNTNTAKSVAILDGVGVNNTFGVISYREKFPMDYHIWHTAPSTGKTYLQPWNDAVEPSTYYIDEGTFLRTGLKFAYIGDNMDLAFTNHTDWKASKHDDGTHRVDLSQRKPANNFMATQRCGKLDPFGYQLKRSKKVFQACGTQVFVGSGRSIDCQWMDSVALNLSRYYGNTEALSSLPLPRNLSADIPAKSSRLFPHGIYNFNFSAPNKRMQRFTASEATTVNGQNQSVYLTYDVPYGRSLYYYTSCALEFRFTNEFFPMDVTPNDGSAQFVVYNMSGNPKKQTTSSKGPTRLYEVARFNCTTRGCEYTQNIPCPRAGHSHTYELAPASPNTSISWIQSYSPRIGVTLQVYSNANFD\n",
      ">G2TRT4\n",
      "MKIANVLTLSFAAIIAASSLAVVVSPITDNSRETASSSISQSPPSQWSKKQLIEYCKKNSLKTSGSHEELVIRVQNHLRTASKKVDARP\n",
      ">A0A0B4GDU5\n",
      "MGWNQTFLFLAFAATAASSLVGSGTSLQLNGIDYFVSPFSQGKVTNGSVAINTRQNQLGFVPATVIAGDLYTESTLQSLFLNWSTVDDVWQPAFLETIFVFNFAKLTNKNHNYHDGVSSSVFPLQVTHKIPSGPYFLNVHTGEVHPAYRLYDDFAGAFTQSLLQRPDGRFQTLSAQVPAAASITIGVPSRLYFTKTEAKPLAGVRIGVKDIFSLAGVKKGCGNRAWYHLYPVANSTGTAMQNLIDKGAIIVGVQKTSQFANGETPTADWVDYHSPFNPRGDGYQDPATSSAGAGSSIASYEWLDLAVGSDTGGSIRGPATVQGIFGNRPSHGLVSLDNVMPLSPKLDTPGFLARDPCLWNAANAALYRDKYTFFGHQAPRYPKKLYLLDFPAGNTSHAPILQNFVTKLAKFLDTSPTNIDLNKEWERTRPTSAGDQSLAQLLNTTYAAIISKDQAKLVREPFYRDYAAVHDGRLPFVNPVPLARWTWGDSQPSSLLSDAVRNKTLFMDWFNGNILPPSSDPLTCSSGLLLHVNGSADFVSRNRYINPPVPPFGFSNSQISLFAETPDSVFPLGQVPVFSSITNNTEYLPVTIDVVAAKGFQLQRDWARLRAILAPALYVDYTKIGKEKWDAMSADDFMAMVSNDDFLGDPCVKTQHLIGATYWERVSESKVIGHHQLRAAHQVYTSPDLKTVKLRGHSHATNEHYYVKSNGVWKFAGLKPEVRWNEYKFEEVFKGSYTQSEKHS\n"
     ]
    }
   ],
   "source": [
    "new_identified = sp_fungi[sp_fungi['Signal peptide'].isna()].copy()\n",
    "new_identified['Entry_'] = '>' + new_identified['Entry']\n",
    "for i, v in new_identified.Entry_.items():\n",
    "    print(\"{}\\n{}\".format(v, new_identified.Protein[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>All_Cleavage</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Q01302</td>\n",
       "      <td>[17]</td>\n",
       "      <td>0.718471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>P87167</td>\n",
       "      <td>[23]</td>\n",
       "      <td>0.744580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7939</th>\n",
       "      <td>G2TRT4</td>\n",
       "      <td>[21]</td>\n",
       "      <td>0.652687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31525</th>\n",
       "      <td>A0A0B4GDU5</td>\n",
       "      <td>[19, 19]</td>\n",
       "      <td>0.834266</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Entry All_Cleavage         Y\n",
       "264        Q01302         [17]  0.718471\n",
       "614        P87167         [23]  0.744580\n",
       "7939       G2TRT4         [21]  0.652687\n",
       "31525  A0A0B4GDU5     [19, 19]  0.834266"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "probable_fungi[probable_fungi.Entry.apply(lambda x: x in new_identified.Entry.values)][['Entry', 'All_Cleavage', 'Y']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Submitted to SignalP 5.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "signalp5 = {\"INFO\":{\"failedjobs\":0,\"size\":4},\"CSV_FILE\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_protein_type.txt\",\"MATURE_FILE\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_mature.fasta\",\"GFF_FILE\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output.gff3\",\"ZIP_FILE\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_all_results.zip\",\"SEQUENCES\":{\"A0A0B4GDU5\":{\"CS_pos\":\"Cleavagesitebetweenpos.19and20:ASS-LV.Probability:0.1952\",\"Likelihood\":[0.9401,0.0599],\"Name\":\"A0A0B4GDU5\",\"Plot_eps\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_A0A0B4GDU5_plot.eps\",\"Plot_png\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_A0A0B4GDU5_plot.png\",\"Plot_txt\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_A0A0B4GDU5_pred.txt\",\"Prediction\":\"Signalpeptide(Sec/SPI)\",\"Protein_types\":[\"SignalPeptide(Sec/SPI)\",\"Other\"]},\"G2TRT4\":{\"CS_pos\":\"Cleavagesitebetweenpos.21and22:SLA-VV.Probability:0.6180\",\"Likelihood\":[0.9506,0.0494],\"Name\":\"G2TRT4\",\"Plot_eps\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_G2TRT4_plot.eps\",\"Plot_png\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_G2TRT4_plot.png\",\"Plot_txt\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_G2TRT4_pred.txt\",\"Prediction\":\"Signalpeptide(Sec/SPI)\",\"Protein_types\":[\"SignalPeptide(Sec/SPI)\",\"Other\"]},\"P87167\":{\"CS_pos\":\"Cleavagesitebetweenpos.23and24:ADA-AP.Probability:0.8428\",\"Likelihood\":[0.9721,0.0279],\"Name\":\"P87167\",\"Plot_eps\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_P87167_plot.eps\",\"Plot_png\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_P87167_plot.png\",\"Plot_txt\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_P87167_pred.txt\",\"Prediction\":\"Signalpeptide(Sec/SPI)\",\"Protein_types\":[\"SignalPeptide(Sec/SPI)\",\"Other\"]},\"Q01302\":{\"CS_pos\":\"Cleavagesitebetweenpos.17and18:AHA-WS.Probability:0.6092\",\"Likelihood\":[0.9975,0.0025],\"Name\":\"Q01302\",\"Plot_eps\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_Q01302_plot.eps\",\"Plot_png\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_Q01302_plot.png\",\"Plot_txt\":\"/services/SignalP-5.0/tmp/5FB19EFE0000792FBC685427/output_Q01302_pred.txt\",\"Prediction\":\"Signalpeptide(Sec/SPI)\",\"Protein_types\":[\"SignalPeptide(Sec/SPI)\",\"Other\"]}},\"FORMAT\":\"long\",\"ORG\":\"Eukarya\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INFO</th>\n",
       "      <th>CSV_FILE</th>\n",
       "      <th>MATURE_FILE</th>\n",
       "      <th>GFF_FILE</th>\n",
       "      <th>ZIP_FILE</th>\n",
       "      <th>SEQUENCES</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>ORG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>failedjobs</th>\n",
       "      <td>0.0</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>size</th>\n",
       "      <td>4.0</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A0A0B4GDU5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>{'CS_pos': 'Cleavagesitebetweenpos.19and20:ASS...</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G2TRT4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>{'CS_pos': 'Cleavagesitebetweenpos.21and22:SLA...</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P87167</th>\n",
       "      <td>NaN</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>{'CS_pos': 'Cleavagesitebetweenpos.23and24:ADA...</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q01302</th>\n",
       "      <td>NaN</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>/services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...</td>\n",
       "      <td>{'CS_pos': 'Cleavagesitebetweenpos.17and18:AHA...</td>\n",
       "      <td>long</td>\n",
       "      <td>Eukarya</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            INFO                                           CSV_FILE  \\\n",
       "failedjobs   0.0  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "size         4.0  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "A0A0B4GDU5   NaN  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "G2TRT4       NaN  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "P87167       NaN  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "Q01302       NaN  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "\n",
       "                                                  MATURE_FILE  \\\n",
       "failedjobs  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "size        /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "A0A0B4GDU5  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "G2TRT4      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "P87167      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "Q01302      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "\n",
       "                                                     GFF_FILE  \\\n",
       "failedjobs  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "size        /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "A0A0B4GDU5  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "G2TRT4      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "P87167      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "Q01302      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "\n",
       "                                                     ZIP_FILE  \\\n",
       "failedjobs  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "size        /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "A0A0B4GDU5  /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "G2TRT4      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "P87167      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "Q01302      /services/SignalP-5.0/tmp/5FB19EFE0000792FBC68...   \n",
       "\n",
       "                                                    SEQUENCES FORMAT      ORG  \n",
       "failedjobs                                                NaN   long  Eukarya  \n",
       "size                                                      NaN   long  Eukarya  \n",
       "A0A0B4GDU5  {'CS_pos': 'Cleavagesitebetweenpos.19and20:ASS...   long  Eukarya  \n",
       "G2TRT4      {'CS_pos': 'Cleavagesitebetweenpos.21and22:SLA...   long  Eukarya  \n",
       "P87167      {'CS_pos': 'Cleavagesitebetweenpos.23and24:ADA...   long  Eukarya  \n",
       "Q01302      {'CS_pos': 'Cleavagesitebetweenpos.17and18:AHA...   long  Eukarya  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sp5 = pd.DataFrame(signalp5)\n",
    "sp5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A0A0B4GDU5    {'CS_pos': 'Cleavagesitebetweenpos.19and20:ASS...\n",
       "G2TRT4        {'CS_pos': 'Cleavagesitebetweenpos.21and22:SLA...\n",
       "P87167        {'CS_pos': 'Cleavagesitebetweenpos.23and24:ADA...\n",
       "Q01302        {'CS_pos': 'Cleavagesitebetweenpos.17and18:AHA...\n",
       "Name: SEQUENCES, dtype: object"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sp5[~sp5['SEQUENCES'].isna()]['SEQUENCES']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Deepsig\n",
    "\n",
    "deepsig prediction results\n",
    "  Copyright (C) 2017 Castrense Savojardo\n",
    "  Department of Biological, Geological and Environmental Sciences (BiGeA)\n",
    "  Bologna Biocomputing Group\n",
    "  University of Bologna, Italy.\n",
    "  savojard@biocomp.unibo.it\n",
    "\n",
    "\n",
    "\n",
    "#### ProteinID\tPredictedSignal\tReliability\tCleavageSite (if any)\n",
    "---------------------------------------------------------------------\n",
    "\n",
    "Q01302\tSignalPeptide\t0.75\t17\n",
    "\n",
    "P87167\tSignalPeptide\t0.99\t23\n",
    "\n",
    "G2TRT4\tSignalPeptide\t0.96\t21\n",
    "\n",
    "A0A0B4GDU5\tSignalPeptide\t0.99\t18\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Razor (SP (Cleavage))</th>\n",
       "      <th>SignalP 5.0 (SP (Cleavage))</th>\n",
       "      <th>DeepSig (SP (Cleavage))</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4GDU5</td>\n",
       "      <td>Yes (19)</td>\n",
       "      <td>Yes (19)</td>\n",
       "      <td>Yes (18)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G2TRT4</td>\n",
       "      <td>Yes (21)</td>\n",
       "      <td>Yes (21)</td>\n",
       "      <td>Yes(21)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P87167</td>\n",
       "      <td>Yes (23)</td>\n",
       "      <td>Yes (23)</td>\n",
       "      <td>Yes (23)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q01302</td>\n",
       "      <td>Yes (17)</td>\n",
       "      <td>Yes (17)</td>\n",
       "      <td>Yes (17)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Entry Razor (SP (Cleavage)) SignalP 5.0 (SP (Cleavage))  \\\n",
       "0  A0A0B4GDU5              Yes (19)                    Yes (19)   \n",
       "1      G2TRT4              Yes (21)                    Yes (21)   \n",
       "2      P87167              Yes (23)                    Yes (23)   \n",
       "3      Q01302              Yes (17)                    Yes (17)   \n",
       "\n",
       "  DeepSig (SP (Cleavage))  \n",
       "0                Yes (18)  \n",
       "1                 Yes(21)  \n",
       "2                Yes (23)  \n",
       "3                Yes (17)  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_results = pd.DataFrame({'Entry':['A0A0B4GDU5', 'G2TRT4', 'P87167', 'Q01302'],\\\n",
    "                             'Razor (SP (Cleavage))':['Yes (19)', 'Yes (21)', 'Yes (23)', 'Yes (17)'],\\\n",
    "                             'SignalP 5.0 (SP (Cleavage))': ['Yes (19)', 'Yes (21)', 'Yes (23)', 'Yes (17)'],\\\n",
    "                             'DeepSig (SP (Cleavage))': ['Yes (18)', 'Yes(21)', 'Yes (23)', 'Yes (17)']})\n",
    "final_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
